src/box/lua/console.lua

```diff
@@ -394,6 +396,85 @@ local function table_pack(...)
     return {n = select('#', ...), ...}
 end
 
+--
+-- In case of errors we set box.error trace frame to the place of box API call
+-- by client (gh-9914). This should be tested. The idea is to use existing diff
+-- tests that use console for test script execution. Let check if trace frame
+-- is correct and if it is not then print the trace frame instead of error
+-- message.
+--
+-- It is hard to fix box.error trace for all API so we will fix it per module
+-- basis. This function checks whether expression is function call of API of
+-- module that have box.error trace fixed.
+--
+local function is_check_trace(expr)
+    if not tarantool.build.test_build then
+        return false
+    end
+    -- First we extract function/functor name from expression.
+    -- Find function name if expression is a an explicit function call.
+    local fun_name = string.match(expr, '^([^(]+)')
+    if fun_name == expr then
+        -- Find function name if expression is a an shortcut function call,
+        -- like f{}.
+        fun_name = string.match(expr, '^([^{]+)')
+    end
+    if fun_name == expr or fun_name == nil then
+        return false
+    end
+    fun_name = string.gsub(fun_name, ':', '.')
+    -- Next get function/functor itself.
+    local fun_getter = loadstring("return " .. fun_name)
+    if not fun_getter then
+        return false
+    end
+    local ok, fun = pcall(fun_getter)
+    if not ok then
+        return false
+    end
+    -- If it is functor then get corresponding function.
+    if type(fun) ~= 'function' then
+        local meta = debug.getmetatable(fun)
+        if meta ~= nil then
+            fun = meta.__call
+        end
+    end
+    if type(fun) ~= 'function' then
+        return false
+    end
+    -- Check the module the function belongs.
+    return utils.proper_trace_required(fun)
```

Comment:
I don't understand what's going on here. Also, some function names are rather unfortunate. We need to discuss this.

Reply:
Approach is reworked to use trace obtained from `xpcall` as discussed so this function is dropped. `proper_trace_required` is renamed to `trace_check_is_required` and moved to `tarantool._internal`.